apis/v1alpha1/targetallocator_types.go

```diff
@@ -93,4 +93,11 @@ type TargetAllocatorSpec struct {
 	// +kubebuilder:default:="30s"
 	// +kubebuilder:validation:Format:=duration
 	CollectorNotReadyGracePeriod *metav1.Duration `json:"collectorNotReadyGracePeriod,omitempty"`
+	// CollectorTargetReloadInterval defines the interval at which the Prometheus receiver will reload targets from the target allocator.
+	// The default is 30s.
+	//
+	// +optional
+	// +kubebuilder:default:="30s"
+	// +kubebuilder:validation:Format:=duration
+	CollectorTargetReloadInterval *metav1.Duration `json:"collectorTargetReloadInterval,omitempty"`
```

Comment:
Do we even need this here? This setting only affects collector behavior, so it should be enough to have it in the collector CRD.